wrapanapi/systems/msazure.py

```diff
@@ -742,6 +743,23 @@ def remove_pips_by_search(self, pip_template, resource_group=None):
             self.logger.debug('No PIPs matching "%s" template were found', pip_template)
             return results
 
+    def remove_discs_by_search(self, disc_name):
+        """
+        Used for clean_up jobs to remove discs that are left after deleting the VM
+        """
+        self.logger.info('Attempting to find the disc image {}'.format(disc_name))
+        results = []
+        path = 'Microsoft.Compute/Images/templates/'
+        discs = self.find_templates(prefix='{}{}'.format(path, disc_name))
+
+        for disc in discs:
+            disc.delete()
+            self.logger.info('disc {} removed'.format(disc_name))
+            results.append(disc_name)
+        if not results:
+            self.logger.debug('No discs matching {} were found'.format(disc_name))
+            return results
```

Comment:
I don't think this return should be indented, we should return results list when it is populated or not.